### PR TITLE
Add CPU feature for stack smash protections

### PR DESCRIPTION
**Pull request by nmeum**
### Contribution description

See https://github.com/RIOT-OS/RIOT/pull/13175#issuecomment-577197439

### Testing procedure

Test with `test/ssp`.

### Issues/PRs references

* #13175

---

CC: @kaspar030

### CODE REVIEW

File: cpu/esp8266/Makefile.features [hunk 1]
```diff
@@ -7,3 +7,4 @@ FEATURES_PROVIDED += periph_cpuid
 FEATURES_PROVIDED += periph_hwrng
 FEATURES_PROVIDED += periph_pm
 FEATURES_PROVIDED += periph_timer
+FEATURES_PROVIDED += ssp
```
Comment (kaspar030) [inline cpu/esp8266/Makefile.features]: @gschorcht ssp doesn't work properly on esp8266, right? (@nmeum it was blacklisted before)

Comment (nmeum) [reply cpu/esp8266/Makefile.features]: It does seem to the pass on the CI, see:

* https://github.com/RIOT-OS/RIOT/pull/13175#issuecomment-577146087
* https://ci.riot-os.org/RIOT-OS/RIOT/13175/4f8f280e1f45301b82b57c84002eb4db97425636/output.html

File: tests/ssp/Makefile [hunk 2]
```diff
@@ -1,7 +1,6 @@
 include ../Makefile.tests_common
 
-# avr8, msp430, esp8266 and mips don't support ssp (yet)
-FEATURES_BLACKLIST += arch_avr8 arch_esp8266 arch_mips32r2 arch_msp430
+FEATURES_REQUIRED += ssp
```
[the remark this answers is not in the document]
Comment (nmeum) [reply tests/ssp/Makefile]: Done

File: cpu/esp8266/Makefile.features (hunk 1 above)
Comment (benpicco) [reply cpu/esp8266/Makefile.features]: esp8266 recently got a toolchain update (https://github.com/RIOT-OS/riotdocker/pull/93), that might have fixed it.

Comment (gschorcht) [reply cpu/esp8266/Makefile.features]: Hm, I hadn't tested it yet for the new platform implementation that uses a newer GCC version. The result of 
```
CONTINUE_ON_EXPECTED_ERRORS=1 make BOARD=esp8266-esp-12x -C tests/ssp/ flash test
```
is
```
START
main(): This is RIOT! (Version: 2020.04-devel-40-g11fee5-periph/gpio/ext_api_proposal_table_void)
calling stack corruption function
*** RIOT kernel panic:
ssp: stack smashing detected
```
So it seems that it works properly now with the new GCC version.

**Comment by kaspar030** [comment]
ACK also from me. &go!

File: cpu/esp8266/Makefile.features (hunk 1 above)
Comment (gschorcht) [reply cpu/esp8266/Makefile.features]: By the way, I guess the problem with SSP on ESP8266 was not related to the old toolchain version. Rather, it was probably caused by the LTO. We had the same problem with ESP32, which I solved by changing `tests/ssp` with PR #12767.

**Pull request merged**